src/ert/storage/local_ensemble.py

```diff
@@ -1214,19 +1158,34 @@ def get_observations_and_responses(
                             f"={response_vals_per_real.shape[1]}"
                         )
 
-                    combined_np_long = np.concatenate(
+                    responses: Dict[str, npt.NDArray[np.float_]] = {
```

Comment:
I think we may leverage `pandas` here and do something like this instead:

```python
responses_df = pd.DataFrame(response_vals_per_real, columns=[str(i) for i in range(response_vals_per_real.shape[1])])

combined_df_long = pd.concat([
    pd.DataFrame({
        "obs_name": obs_names_1d,
        "key_index": key_index_1d,
        "observations": obs_vals_1d,
        "errors": std_vals_1d,
    }),
    responses_df.astype('float32')
], axis=1)
```

Does this make sense you think?